### ISSUE TITLE
SALT.MODULES.WIN_PKG - UnicodeEncodeError

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#45886]
### Description of Issue/Question
Trying to run the `salt.modules.win_pkg` fails on `oxygen.rc1`.
See this:
```
    The minion function caused an exception: Traceback (most recent call last):
      File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1624, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "c:\salt\bin\lib\site-packages\salt\executors\direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1142, in install
        old = list_pkgs(saltenv=saltenv, refresh=refresh, versions_as_list=True)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 400, in list_pkgs
        name_map = _get_name_map(saltenv)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1846, in _get_name_map
        u_name_map[k.decode('utf-8')] = name_map[k]
      File "c:\salt\bin\lib\encodings\utf_8.py", line 16, in decode
        return codecs.utf_8_decode(input, errors, True)
    UnicodeEncodeError: 'ascii' codec can't encode characters in position 5-11: ordinal not in range(128)
```

### Setup
WIndows minion and linux master.

### Steps to Reproduce Issue
Try to run `salt.modules.win_pkg`commands from the master to the win minion and it will fail.
```
[root@localhost ~]# salt '*' pkg.install firefox
win:
    The minion function caused an exception: Traceback (most recent call last):
      File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1624, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "c:\salt\bin\lib\site-packages\salt\executors\direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1142, in install
        old = list_pkgs(saltenv=saltenv, refresh=refresh, versions_as_list=True)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 400, in list_pkgs
        name_map = _get_name_map(saltenv)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1846, in _get_name_map
        u_name_map[k.decode('utf-8')] = name_map[k]
      File "c:\salt\bin\lib\encodings\utf_8.py", line 16, in decode
        return codecs.utf_8_decode(input, errors, True)
    UnicodeEncodeError: 'ascii' codec can't encode characters in position 5-11: ordinal not in range(128)
[root@localhost ~]# salt '*' pkg.list_pkgs
win:
    The minion function caused an exception: Traceback (most recent call last):
      File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1624, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "c:\salt\bin\lib\site-packages\salt\executors\direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 400, in list_pkgs
        name_map = _get_name_map(saltenv)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1846, in _get_name_map
        u_name_map[k.decode('utf-8')] = name_map[k]
      File "c:\salt\bin\lib\encodings\utf_8.py", line 16, in decode
        return codecs.utf_8_decode(input, errors, True)
    UnicodeEncodeError: 'ascii' codec can't encode characters in position 5-11: ordinal not in range(128)
[root@localhost ~]# salt '*' pkg.latest_version chrome
win:
    The minion function caused an exception: Traceback (most recent call last):
      File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1624, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "c:\salt\bin\lib\site-packages\salt\executors\direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 128, in latest_version
        versions_as_list=True, saltenv=saltenv, refresh=refresh)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 400, in list_pkgs
        name_map = _get_name_map(saltenv)
      File "c:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1846, in _get_name_map
        u_name_map[k.decode('utf-8')] = name_map[k]
      File "c:\salt\bin\lib\encodings\utf_8.py", line 16, in decode
        return codecs.utf_8_decode(input, errors, True)
    UnicodeEncodeError: 'ascii' codec can't encode characters in position 5-11: ordinal not in range(128)
[root@localhost ~]# 
```

### Versions Report
```
    Salt Version:
               Salt: 2018.2.0-227-gfcf8a36
     
    Dependency Versions:
               cffi: 1.10.0
           cherrypy: Not Installed
           dateutil: 2.6.1
          docker-py: Not Installed
              gitdb: 2.0.3
          gitpython: 2.1.7
              ioflo: Not Installed
             Jinja2: 2.10
            libgit2: Not Installed
            libnacl: 1.6.1
           M2Crypto: Not Installed
               Mako: 1.0.7
       msgpack-pure: Not Installed
     msgpack-python: 0.4.8
       mysql-python: Not Installed
          pycparser: 2.18
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 2.7.14 (v2.7.14:84471935ed, Sep 16 2017, 20:25:58) [MSC v.1500 64 bit (AMD64)]
       python-gnupg: 0.4.1
             PyYAML: 3.12
              PyZMQ: 16.0.3
               RAET: Not Installed
              smmap: 2.0.3
            timelib: 0.2.4
            Tornado: 4.5.2
                ZMQ: 4.1.6
     
    System Versions:
               dist:   
             locale: cp1252
            machine: AMD64
            release: 2016Server
             system: Windows
            version: 2016Server 10.0.14393  Multiprocessor Free
```

**Comment by rallytime** [comment]
@twangboy Can you look into this?

**Comment by damon-atkins** [comment]
Try pkg.refresh_db first, although it should have done this.  Can you install PY3 please and try it as well.

**Comment by frogunder** [comment]
On a py3 setup it is working correctly, so it seems it is only failing on a py2 setup.

Running pkg.refresh_db command worked but it I still get the error when trying to run other commands.

```
[root@localhost ~]# salt '*' pkg.refresh_db
win-rename:
    ----------
    failed:
        0
    success:
        251
    total:
        251
[root@localhost ~]# salt '*' pkg.list_pkgs
win-rename:
    The minion function caused an exception: Traceback (most recent call last):
      File "C:\salt\bin\lib\site-packages\salt\minion.py", line 1624, in _thread_return
        return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)
      File "C:\salt\bin\lib\site-packages\salt\executors\direct_call.py", line 12, in execute
        return func(*args, **kwargs)
      File "C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 400, in list_pkgs
        name_map = _get_name_map(saltenv)
      File "C:\salt\bin\lib\site-packages\salt\modules\win_pkg.py", line 1846, in _get_name_map
        u_name_map[k.decode('utf-8')] = name_map[k]
      File "C:\salt\bin\lib\encodings\utf_8.py", line 16, in decode
        return codecs.utf_8_decode(input, errors, True)
    UnicodeEncodeError: 'ascii' codec can't encode characters in position 5-11: ordinal not in range(128)
[root@localhost ~]#
```

Running minion in debug mode, this is what I see:
```
[INFO    ] User root Executing command pkg.list_pkgs with jid 20180207153603173794
[DEBUG   ] Command details {u'tgt_type': u'glob', u'jid': u'20180207153603173794', u'tgt': u'*', u'ret': u'', u'user': u'root', u'arg': [], u'fun': u'pkg.list_pkgs'}
[DEBUG   ] Multiprocessing queue logging configured for the process running under PID: 4920
[DEBUG   ] Grains refresh requested. Refreshing grains.
[DEBUG   ] Reading configuration from c:\salt\conf\minion
[DEBUG   ] Including configuration from 'c:\salt\conf\minion.d\_schedule.conf'
[DEBUG   ] Reading configuration from c:\salt\conf\minion.d\_schedule.conf
[INFO    ] Executing command 'Powershell -NonInteractive -NoProfile "try {Get-WmiObject -ErrorAction Stop -class MSFC_FibrePortHBAAttributes -namespace \"root\WMI\" | Select -Expandproperty Attributes | %{($_.PortWWN | % {\"{0:x2}\" -f $_}) -join \"\"} | ConvertTo-JSON} catch { \"{}\" | ConvertTo-JSON}"' in directory 'C:\Users\Administrator'
[DEBUG   ] output: "{}"
[DEBUG   ] Motherboard info not available on this system
[INFO    ] Creating minion process manager
[INFO    ] Starting a new job with PID 4920
[DEBUG   ] LazyLoaded pkg.list_pkgs
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Using existing pkg metadata db for saltenv 'base' (age is 0:01:40.389843)
[INFO    ] User root Executing command saltutil.find_job with jid 20180207153608195908
[DEBUG   ] Command details {u'tgt_type': u'list', u'jid': u'20180207153608195908', u'tgt': [u'win-rename'], u'ret': u'', u'user': u'root', u'arg': [u'20180207153603173794'], u'fun': u'saltutil.find_job'}
[WARNING ] The minion function caused an exception
[DEBUG   ] SaltEvent PUB socket URI: 4510
[DEBUG   ] SaltEvent PULL socket URI: 4511
[DEBUG   ] Initializing new IPCClient for path: 4511
[DEBUG   ] Sending event: tag = _salt_error; data = {'message': u'The minion function caused an exception', 'args': ('The minion function caused an exception',), u'_stamp': '2018-02-07T15:36:08.300000'}
[DEBUG   ] Minion return retry timer set to 10 seconds (randomized)
[INFO    ] Returning information for job: 20180207153603173794
[DEBUG   ] Minion of 'frodec7' is handling event tag '_salt_error'
[DEBUG   ] Forwarding salt error event tag=_salt_error
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506', u'aes')
[DEBUG   ] Initializing new AsyncAuth for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://10.1.51.103:4506
[DEBUG   ] Trying to connect to: tcp://10.1.51.103:4506
[DEBUG   ] Popen(['git', 'version'], cwd=C:\salt, universal_newlines=False, shell=None)
[DEBUG   ] Popen(['git', 'version'], cwd=C:\salt, universal_newlines=False, shell=None)
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506', u'aes')
[DEBUG   ] Initializing new AsyncAuth for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://10.1.51.103:4506
[DEBUG   ] Trying to connect to: tcp://10.1.51.103:4506
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506', 'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://10.1.51.103:4506
[DEBUG   ] Trying to connect to: tcp://10.1.51.103:4506
[DEBUG   ] salt.crypt.get_rsa_pub_key: Loading public key
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] salt.crypt.get_rsa_key: Loading private key
[DEBUG   ] salt.crypt._get_key_with_evict: Loading private key
[DEBUG   ] Loaded minion key: c:\salt\conf\pki\minion\minion.pem
[DEBUG   ] salt.crypt.get_rsa_pub_key: Loading public key
[DEBUG   ] minion return: {u'fun_args': [], u'jid': u'20180207153603173794', u'return': u'The minion function caused an exception: Traceback (most recent call last):\n  File "C:\\salt\\bin\\lib\\site-packages\\salt\\minion.py", line 1624, in _thread_return\n    return_data = minion_instance.executors[fname](opts, data, func, args, kwargs)\n  File "C:\\salt\\bin\\lib\\site-packages\\salt\\executors\\direct_call.py", line 12, in execute\n    return func(*args, **kwargs)\n  File "C:\\salt\\bin\\lib\\site-packages\\salt\\modules\\win_pkg.py", line 400, in list_pkgs\n    name_map = _get_name_map(saltenv)\n  File "C:\\salt\\bin\\lib\\site-packages\\salt\\modules\\win_pkg.py", line 1846, in _get_name_map\n    u_name_map[k.decode(\'utf-8\')] = name_map[k]\n  File "C:\\salt\\bin\\lib\\encodings\\utf_8.py", line 16, in decode\n    return codecs.utf_8_decode(input, errors, True)\nUnicodeEncodeError: \'ascii\' codec can\'t encode characters in position 5-11: ordinal not in range(128)\n', u'success': False, u'fun': u'pkg.list_pkgs', u'out': u'nested'}
[DEBUG   ] Multiprocessing queue logging configured for the process running under PID: 2844
[DEBUG   ] Grains refresh requested. Refreshing grains.
[DEBUG   ] Reading configuration from c:\salt\conf\minion
[DEBUG   ] Including configuration from 'c:\salt\conf\minion.d\_schedule.conf'
[DEBUG   ] Reading configuration from c:\salt\conf\minion.d\_schedule.conf
[INFO    ] Executing command 'Powershell -NonInteractive -NoProfile "try {Get-WmiObject -ErrorAction Stop -class MSFC_FibrePortHBAAttributes -namespace \"root\WMI\" | Select -Expandproperty Attributes | %{($_.PortWWN | % {\"{0:x2}\" -f $_}) -join \"\"} | ConvertTo-JSON} catch { \"{}\" | ConvertTo-JSON}"' in directory 'C:\Users\Administrator'
[DEBUG   ] output: "{}"
[DEBUG   ] Motherboard info not available on this system
[INFO    ] Creating minion process manager
[INFO    ] Starting a new job with PID 2844
[DEBUG   ] LazyLoaded saltutil.find_job
[DEBUG   ] LazyLoaded direct_call.execute
[DEBUG   ] Minion return retry timer set to 8 seconds (randomized)
[INFO    ] Returning information for job: 20180207153608195908
[DEBUG   ] Popen(['git', 'version'], cwd=C:\salt, universal_newlines=False, shell=None)
[DEBUG   ] Popen(['git', 'version'], cwd=C:\salt, universal_newlines=False, shell=None)
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506', u'aes')
[DEBUG   ] Initializing new AsyncAuth for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://10.1.51.103:4506
[DEBUG   ] Trying to connect to: tcp://10.1.51.103:4506
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'c:\\salt\\conf\\pki\\minion', 'win-rename', u'tcp://10.1.51.103:4506', 'clear')
[DEBUG   ] Connecting the Minion to the Master URI (for the return server): tcp://10.1.51.103:4506
[DEBUG   ] Trying to connect to: tcp://10.1.51.103:4506
[DEBUG   ] salt.crypt.get_rsa_pub_key: Loading public key
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] salt.crypt.get_rsa_key: Loading private key
[DEBUG   ] salt.crypt._get_key_with_evict: Loading private key
[DEBUG   ] Loaded minion key: c:\salt\conf\pki\minion\minion.pem
[DEBUG   ] salt.crypt.get_rsa_pub_key: Loading public key
[DEBUG   ] minion return: {u'fun_args': [u'20180207153603173794'], u'jid': u'20180207153608195908', u'return': {}, u'retcode': 0, u'success': True, u'fun': u'saltutil.find_job'}
```

**Comment by twangboy** [comment]
This is an issue with one of the package definition files. It has a unicode character.

**Comment by damon-atkins** [comment]
The old reg code use to handle it better or ignore something.
@thatch45    However it highlights again PY2 issues on Windows.
i.e. 2017.11 use PY2 recommanded on windows
2018 onwards PY3 recommended on windows.

**Comment by rallytime** [comment]
This is fixed with #45937
